src/sentry/workflow_engine/migrations/0059_fix_high_priority_condition_triggers.py

```diff
@@ -0,0 +1,110 @@
+import logging
+
+from django.apps.registry import Apps
+from django.conf import settings
+from django.db import migrations, router, transaction
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils import redis
+from sentry.utils.iterators import chunked
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBarApprox
+
+logger = logging.getLogger(__name__)
+
+CHUNK_SIZE = 1000
+
+
+def move_high_priority_conditions_to_triggers(
+    apps: Apps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    DataCondition = apps.get_model("workflow_engine", "DataCondition")
+    Workflow = apps.get_model("workflow_engine", "Workflow")
+    WorkflowDataConditionGroup = apps.get_model("workflow_engine", "WorkflowDataConditionGroup")
+
+    backfill_key = "workflow_engine_fix_high_priority_conditions"
+    redis_client = redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
+    progress_id = int(redis_client.get(backfill_key) or 0)
+
+    for workflows in chunked(
+        RangeQuerySetWrapperWithProgressBarApprox(
+            Workflow.objects.filter(id__gt=progress_id), step=CHUNK_SIZE
+        ),
+        CHUNK_SIZE,
+    ):
+        for workflow in workflows:
+            with transaction.atomic(router.db_for_write(Workflow)):
+                current_workflow = (
+                    Workflow.objects.select_for_update().filter(id=workflow.id).first()
+                )
+
+                if not current_workflow:
+                    continue
+
+                when_condition_group_id = current_workflow.when_condition_group_id
+                workflow_dcg = WorkflowDataConditionGroup.objects.filter(
+                    workflow_id=workflow.id
```

Comment:
nit: it's a little confusing to mix "current_workflow" and "workflow" -- it seems like they're the same?